odoo/addons/base/models/res_partner_bank.py

```diff
@@ -0,0 +1,208 @@
+import re
+
+from odoo import api, fields, models
+from odoo.exceptions import ValidationError
+
+
+def sanitize_account_number(acc_number):
+    if acc_number:
+        return re.sub(r'\W+', '', acc_number).upper()
+    return False
+
+
+class ResPartnerBank(models.Model):
+    _name = 'res.partner.bank'
+    _rec_name = 'acc_number'
+    _description = 'Bank Account'
+    _order = 'sequence, id'
+
+    @api.model
+    def get_supported_account_types(self):
+        return self._get_supported_account_types()
+
+    @api.model
+    def _get_supported_account_types(self):
+        return [('bank', self.env._('Normal'))]
+
+    active = fields.Boolean(default=True)
+    acc_type = fields.Selection(selection=lambda x: x.env['res.partner.bank'].get_supported_account_types(), compute='_compute_acc_type', string='Type', help='Bank account type: Normal or IBAN. Inferred from the bank account number.')
+    acc_number = fields.Char('Account Number', required=True, search='_search_acc_number')
+    sanitized_acc_number = fields.Char(compute='_compute_sanitized_acc_number', string='Sanitized Account Number', readonly=True, store=True)
+    acc_holder_name = fields.Char(string='Holder Name', help="Account holder name, in case it is different than the name of the Account Holder", compute='_compute_account_holder_name', readonly=False, store=True)
+    partner_id = fields.Many2one('res.partner', "Partner", ondelete='cascade', index=True, domain=['|', ('is_company', '=', True), ('parent_id', '=', False)], required=True)
+    allow_out_payment = fields.Boolean('Send Money', help='This account can be used for outgoing payments', default=False, copy=False, readonly=False)
+
+    # bank fields
```

Comment:
Remove the "bank_" prefix. We are already on a bank object and it will avoid some helpers you have in the business code (using bank_prefix).